pkg/operator/ceph/object/objectstore.go

```diff
@@ -351,3 +354,76 @@ func GetObjectBucketProvisioner(c *clusterd.Context, namespace string) string {
 	}
 	return provName
 }
+
+func checkDashboardUser(context *Context) bool {
+	args := []string{"dashboard", "get-rgw-api-access-key"}
+	cephCmd := ceph.NewCephCommand(context.Context, context.ClusterName, args)
+	out, _ := cephCmd.Run()
+	if string(out) == "" {
+		return false
+	} else {
+		return true
+	}
+}
+
+func enableRGWDashboard(context *Context) error {
+	if checkDashboardUser(context) {
+		logger.Debug("Dash board already enabled for rgw")
+		return nil
+	}
+	//currently adminuser only for accessing dashboard, in future this will be seperate api
+	dispname := fmt.Sprintf("%s", AdminUser)
+	user := ObjectUser{
+		UserID:      AdminUser,
+		DisplayName: &dispname,
+		SystemUser:  true,
+	}
+	u, errCode, err := CreateUser(context, user)
+	if err != nil || errCode != 0 {
+		return errors.Wrap(err, "error creating admin user")
+	}
+	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
+	cephCmd := ceph.NewCephCommand(context.Context, context.ClusterName, args)
+	_, err = cephCmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "error setting user accesskey")
+	}
+
+	args = []string{"dashboard", "set-rgw-api-secret-key", *u.SecretKey}
+	cephCmd = ceph.NewCephCommand(context.Context, context.ClusterName, args)
+	_, err = cephCmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "error setting user secretkey")
+
+	}
+
+	return nil
+}
+func disableRGWDashboard(context *Context) error {
```

Comment:
Creation and deletions are often done in opposite orders. Should delete remove the user after resetting keys, or before?

Reply:
@BlaineEXE  AFAIK resetting does not need the user to be existent so both way should work IMO